lib/cache-utils.js

```diff
@@ -55,25 +55,26 @@ function wellKnownCachePath() {
 
 export async function getCacheKey() {
   const platform = process.env.RUNNER_OS;
-  const fileHash = await glob.hashFiles(await getScarbManifestPath());
+  const fileHash = await glob.hashFiles(await getScarbLockfilePath());
 
   if (!fileHash) {
     throw new Error(
-      "failed to cache dependencies: unable to hash Scarb.toml file",
+      "failed to cache dependencies: unable to hash Scarb.lock file",
     );
   }
 
   return `scarb-cache-${platform}-${fileHash}`.toLowerCase();
 }
 
-async function getScarbManifestPath() {
-  const { stdout, exitCode } = await exec.getExecOutput("scarb manifest-path");
+async function getScarbLockfilePath() {
+  const globber = await glob.create("**/Scarb.lock");
+  const lockfiles = await globber.glob();
 
-  if (exitCode > 0) {
-    throw new Error(
-      "failed to find Scarb.toml: command `scarb manifest-path` failed",
-    );
+  if (lockfiles.length === 0) {
+    throw new Error("failed to find Scarb.lock");
   }
 
-  return stdout.trim();
+  return lockfiles.reduce((prev, next) =>
+    prev.length < next.length ? prev : next,
```

Comment:
For a project structure 
```
Scarb.lock
A ->
      | A -> 
               |  Scarb.lock
scarb -> 
               | is ->  
                          | superb ->
                                              | Scarb.lock
```
When run in `/scarb/is/superb/`, what will it return? 🤔